site/zenodo_rdm/assets/semantic-ui/js/zenodo_rdm/src/support/components/SupportForm.js

```diff
@@ -0,0 +1,290 @@
+                        <label className="helptext mt-0">
+                            {`Optional. Max attachments size:`} {humanReadableBytes(maxFileSize, false)}
+                        </label>
+                        {currentFiles.length > 0
+                            &&
+                            <FileTable
+                                handleDelete={handleDelete}
+                                filesList={currentFiles}
+                            />
+                        }
+                    </div>
+                )
+            }}
+        </Dropzone>
+    );
+};
+
+FileUploader.propTypes = {
+    dropzoneParams: PropTypes.object.isRequired,
+    maxFileSize: PropTypes.number.isRequired,
+    name: PropTypes.string.isRequired,
+    currentFiles: PropTypes.array.isRequired
+};
+
+class SupportForm extends Component {
+
+    constructor(props) {
+        super(props);
+        this.FormSchema = Yup.object({
+            name: Yup.string().required(),
+            email: Yup.string().email().required(),
+            category: Yup.string().required(),
+            subject: Yup.string().required(),
+            description: Yup.string().min(
+                this.props.descriptionMinLength,
+                `Field must be at least ${this.props.descriptionMinLength} characters long.`
+            ).max(
+                this.props.descriptionMaxLength,
+                `Field must be at most ${this.props.descriptionMaxLength} characters long.`)
+                .required(),
+            sysInfo: Yup.boolean()
+        });
+
+        this.state = {
+            files: []
+        };
+    }
+
+    onSubmit = async (values, formikBag) => {
+        const { apiEndpoint } = this.props;
+        let response;
+        const formData = formikToFormData(values);
+        try {
+            response = await http.post(apiEndpoint, formData);
+        } catch (error) {
+            let errMessage = error.message;
+            if (response) {
+                errMessage = error.response.data;
+            }
+            
+            // TODO show error notification
+            console.error(errMessage); 
+        }
+
+        // TODO show a success notification
+        window.location = response.request.responseURL;
+    }
+
+    updateFiles = (files, formikSetFieldValue) => {
+        formikSetFieldValue("files", files)
+        this.setState({
+            files: files
+        });
+    }
+
+    render() {
+        const {
+            categories,
+            name,
+            isUserAuthenticated,
+            userBrowser,
+            userPlatform,
+            userMail,
+            maxFileSize
+        } = this.props;
+
+        const defaultCategory = categories.length > 0 ? categories[0].key : '';
+
+        const initialValues = {
+            email: userMail,
+            name: name,
+            category: defaultCategory,
+            sysInfo: false,
+            files: null
+        }
```

Comment:
does it make sense to be moved to the constructor instead of redeclaring it on each render?